ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credentials/new.js

```diff
@@ -19,15 +19,24 @@ export default class ScopesScopeCredentialStoresCredentialStoreCredentialsNewRou
 
   /**
    * Creates a new unsaved credential in current credential store.
+   * Also rollback/destroy any new, unsaved instances from this route before
+   * creating another, but reuse name/description if available.
    * @return {CredentialModel}
    */
   model({ type = 'username_password' }) {
     const { id: credential_store_id } = this.modelFor(
       'scopes.scope.credential-stores.credential-store'
     );
+    let name, description;
+    if (this.currentModel?.isNew) {
+      ({ name, description } = this.currentModel);
```

Comment:
This is clever syntax, I haven't seen destructuring to existing variables before. 🤯